GitOut/Features/Git/Log/TreeBuildingLeaf.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace GitOut.Features.Git.Log
+{
+    public class TreeBuildingLeaf
+    {
+        public TreeBuildingLeaf(GitHistoryEvent? lookingFor, GitTreeNode currentNode)
+        {
+            LookingFor = lookingFor;
+            Current = currentNode;
+        }
+
+        public GitHistoryEvent? LookingFor { get; }
+        public GitTreeNode Current { get; }
+    }
```

Comment:
Instead of passing null to this constructor, use static factory method
```suggestion
        private TreeBuildingLeaf(GitHistoryEvent? lookingFor, GitTreeNode currentNode)
        {
            LookingFor = lookingFor;
            Current = currentNode;
        }

        public GitHistoryEvent? LookingFor { get; }
        public GitTreeNode Current { get; }

        public static TreeBuildingLeaf WithoutParent(GitTreeNode node) => new TreeBuildingLeaf(null, node);

        public static TreeBuildingLeaf WithParent(GitHistoryEvent parent, GitTreeNode node) => new TreeBuildingLeaf(parent, node);
    }
```
That way we will get null validation checks